apps/frontend/src/components/global/ExportCaat.vue

```diff
@@ -38,190 +33,24 @@ type CAATRow = {
-        hdf.wraps.title
-      )}`;
-      if (hdf.start_time) {
-        row['Date Identified'] = this.convertDate(
-          new Date(hdf.start_time),
-          '/'
-        );
-      }
-      row['Finding ID'] = `${filename} - Test ${this.fix(hdf.wraps.id)}`;
-      row['Finding Description'] = this.fix(hdf.wraps.title);
-      row['Weakness Description'] = this.createCaveat(hdf);
-      row['Control Weakness Type'] = 'Security';
-      row['Source'] = 'Self-Assessment';
-      row['Test Method'] = 'Test';
-      row['Test Objective'] = this.fix(
-        hdf.descriptions.check || hdf.wraps.tags.check
-      );
-      row['Test Result Description'] = this.fix(
-        this.createTestResultDescription(hdf)
-      );
-      row['Test Result'] = this.createTestResult(hdf);
-      row['Recommended Corrective Action(s)'] = this.fix(
-        hdf.descriptions.fix || hdf.wraps.tags.fix
-      );
-      row['Impact'] = this.createImpact(hdf);
-      allRows.push(row);
-    }
-    return allRows;
-  }
-
   exportCaat() {
-    // Define our workbook
-    const wb = XLSX.utils.book_new();
-    // For each file in our filter
-    this.filter.fromFile.forEach((fileId) => {
-      // Find our file within InspecDataModule
-      const file = InspecDataModule.allFiles.find((f) => f.uniqueId === fileId);
-      let renameCount = 2;
-      let sheetName: string = `${file?.filename || fileId}`.substring(0, 31);
-      while (this.takenFilenames.includes(sheetName)) {
-        sheetName =
-          `${file?.filename || fileId} `.substring(0, 26) + ` (${renameCount})`;
-        renameCount++;
-      }
-      this.takenFilenames.push(sheetName);
-      // Create a new Sheet
-      wb.SheetNames.push(sheetName);
-      wb.Props = this.fileSettings;
-      // Get the controls for the current file
+    const inputData = this.filter.fromFile.map((fileId: string) => {
+      const file = (
+        InspecDataModule.allEvaluationFiles as EvaluationFile[]
+      ).find((f) => f.uniqueId === fileId);
+      const data = file?.evaluation ?? '';
+      const filename = file?.filename || fileId;
       const controls = FilteredDataModule.controls({
         ...this.filter,
         fromFile: [fileId]
-      });
-      const hitIds = new Set();
-      const rows: CAATRow[] = [];
-      // Convert them into rows
-      for (const ctrl of controls) {
-        const root = ctrl.root;
-        if (hitIds.has(root.hdf.wraps.id)) {
-          continue;
-        } else {
-          hitIds.add(root.hdf.wraps.id);
-          rows.push(...this.getRow(root, file?.filename));
-        }
-      }
-      // Add rows to sheet
-      const ws = XLSX.utils.json_to_sheet(rows, {header: this.header});
-      wb.Sheets[sheetName] = ws;
+      }).slice();
+      return {data, filename, controls};
     });
-
-    const wbout = XLSX.write(wb, {bookType: 'xlsx', type: 'binary'});
+    const caat = new FromHDFToCAATMapper(inputData).toCAAT(false);
```

Comment:
do you want to pass returnWorkbook=false for readability in .toCAAT?

Reply:
I don't know if that's necessary since they can just use their ide to figure out the parameter name/type which clearly (hopefully) states what's going on